README.md
# EJO Base Module Manager
This plugin manages the modules which can be enabled or disabled based on the clients plan


CHANGES.md
- Simplified unregistering of widgets
- Improved blog-module and removed contactads module
- Added ejo_add_extended_theme_support function as a helper

0.1
- Named plugin EJO Base Module Manager
- Migrated Module Manager from EJO Base

includes/helpers.php
<?php

/* Function might already be included by EJO Base */
if (! function_exists('ejo_add_extended_theme_support')) {
	/**
	 * Wordpress only accepts arguments at 'current_theme_supports' function for
	 * it's own built-in functions. This function can be used for filtering custom 
	 * theme supported functions
	 *
	 * Use:
	 * add_filter( 'current_theme_supports-feature', 'ejo_add_extended_theme_support', 10, 3 );
	 *
	 * Then use in theme:
	 * add_theme_support( 'feature', array( 'item1', 'item2' ) );
	 *
	 * And check in plugin:
	 * - require_if_theme_supports( 'feature', path );
	 * - if (current_theme_supports( 'feature', 'item1' ))
	 */
	function ejo_add_extended_theme_support($extended_theme_support = true, $checked_arg = array(), $theme_support_args = false)
	{
		//* If array: Check if feature-option is supported by theme
		if ( is_array($theme_support_args) ) {		
			$extended_theme_support = in_array( $checked_arg[0], $theme_support_args[0] );
		}

	    return $extended_theme_support;
	}
}

includes/class-module.php
<?php

/**
 * Class to eextend
 */
class EJO_Base_Module
{
    /* Only instantiate once */
    public static function instance() 
    {
        if ( !self::$_instance )
            self::$_instance = new self;
        return self::$_instance;
    }

    //* No cloning
    protected function __clone() {}

    /* Plugin setup. */
    protected function __construct() {}

    /**
     * Get Module
     */
    public static function get_module( $id )
    {
        $modules = EJO_Base_Module_Manager::$modules;

        //* Return false if module id is not found
        if ( empty($modules[$id]) )
            return false;

        return $modules[$id];
    }

    /**
     * Get name
     */
    public static function get_name( $id )
    {
        $module = self::get_module( $id );

        return $module['name'];
    }

    /**
     * Get description
     */
    public static function get_description( $id )
    {
        $module = self::get_module( $id );

        if ( empty($module['description']) )
            return array();

        return $module['description'];
    }

    /**
     * Get dependancies
     */
    public static function get_dependancies( $id )
    {
        $module = self::get_module( $id );

        if ( empty($module['dependancies']) )
            return array();

        return $module['dependancies'];
    }


    //* Check if module has theme support
    public static function has_theme_support( $id ) 
    {
        //* Return true if theme supports the module
        if ( current_theme_supports( 'ejo-base-modules', $id ) )
            return true;

        return false;
    }

    /**
     * Check if module exists
     */
    public static function exists( $id ) 
    {
        return ( self::get_module( $id ) != false );
    }

    /**
     * Check if module is available
     */
    public static function is_available( $id ) 
    {
        //* Check if current theme supports the module
        if ( self::exists($id) && self::has_theme_support($id) && self::has_no_missing_dependancies($id) ) {
            return true;
        }
        else {
            return false;
        }
    }

    /**
     * Check if module is stored as active in database
     * 
     * is_active function checks wether module is_active and available
     */
    public static function is_activated( $id, $active_modules = null ) 
    {
        if ( ! $active_modules )
            $active_modules = get_option( 'ejo_base_active_modules', array() );
        
        //* Check if module is active in options
        if ( in_array($id, $active_modules) )
            return true;

        return false;
    }

    /**
     * Check if module is stored as active in database and if it is available
     */
    public static function is_active( $id, $active_modules = null ) 
    {
        if ( ! $active_modules )
            $active_modules = get_option( 'ejo_base_active_modules', array() );
        
        //* Check if module is active in options
        if ( self::is_activated($id, $active_modules) && self::is_available($id) )
            return true;

        return false;
    }

    /**
     * Wrapper for is_active
     */
    public static function is_inactive( $id, $active_modules = null ) 
    {
        if ( ! $active_modules )
            $active_modules = get_option( 'ejo_base_active_modules', array() );
        
        return ! self::is_active( $id, $active_modules );
    }


    public static function has_missing_dependancies( $id ) 
    {
        //* Get missing dependancies
        $missing_dependancies = self::get_missing_dependancies($id);

        //* If no missing dependancies, return false
        if ( empty($missing_dependancies) )
            return false;

        //* Module has missing dependancies
        return true;
    }

    public static function has_no_missing_dependancies( $id ) 
    {
        return ! self::has_missing_dependancies($id);
    }

    public static function check_dependancies( $dependancies )
    {
        $missing_dependancies = array();

        //* Check if dependancies are fulfilled
        foreach ($dependancies as $dependancy) {

            if ($dependancy['type'] == 'ejo-base-module') {

                $dependant_module = self::get_module($dependancy['id']);

                if ( ! self::is_active($dependant_module['id']) ) {
                    $missing_dependancies[] = $dependant_module['name'];
                }
            }
            elseif ( $dependancy['type'] == 'plugin' ) {

                $dependant_plugin = $dependancy;

                if ( ! class_exists( $dependant_plugin['class'] ) ) {
                    $missing_dependancies[] = $dependant_plugin['name'];
                }
            }
        }

        return $missing_dependancies;
    }

    public static function get_missing_dependancies( $id )
    {
        $dependancies = self::get_dependancies($id);

        return self::check_dependancies( $dependancies );
    }

    
    public static function show_dependancies( $id )
    {
        $dependancies = self::get_dependancies( $id );
           
        if ( empty($dependancies) )
            return;

        $output = '';

        // $output = __( 'Dependancies: ');

        foreach ($dependancies as $dependancy) {

            if ($dependancy['type'] == 'ejo-base-module') {

                $dependant_module = self::get_module($dependancy['id']);

                if ( self::is_active($dependant_module['id']) ) {
                    $output .= '<span class="dependancy-good">' . $dependant_module['name'] . ': ' . __( 'active') . ' |</span> ';
                }
                else {
                    $output .= $dependant_module['name'] . ': ' . __( 'inactive') . ' | ';
                }

            }
            elseif ( $dependancy['type'] == 'plugin' ) {

                $dependant_plugin = $dependancy;

                if ( class_exists( $dependant_plugin['class'] ) ) {
                    $output .= '<span class="dependancy-good">' . $dependant_plugin['name'] . ': ' . __( 'active') . ' |</span> ';
                }
                else {
                    $output .= $dependant_plugin['name'] . ': ' . __( 'inactive') . ' | ';
                }
            }
        }

        //* Remove ` |</span> ` and ` | ` from end of output
        $output = preg_replace('/ \|(<\/span>)? $/', '', $output);

        if ( ! self::has_theme_support($id) || self::has_no_missing_dependancies($id) ){
            $output = '<div class="dependancies-hidden">' . $output . '</div>';
        }

        echo $output;
    }

    /**
     * Show why module is not available
     */
    public static function print_why_not_available( $id ) 
    {
        if ( ! self::exists($id) ) {
            echo __( 'Module doesn\'t exist', EJO_Base_Module_Manager::$slug );
        }
        elseif ( ! self::has_theme_support($id) ) {
            echo __( 'No theme-support', EJO_Base_Module_Manager::$slug );
        }
        elseif ( self::has_missing_dependancies($id) ) {
            echo __( 'Missing dependancies', EJO_Base_Module_Manager::$slug );
        }
        else {
            echo __( 'Module not available', EJO_Base_Module_Manager::$slug );
        }
    }

    public static function activate( $id, $active_modules = null )
    {
        if ( ! $active_modules )
            $active_modules = get_option( 'ejo_base_active_modules', array() );

        //* Store module if it doesn't already exist
        if ( !self::is_activated( $id, $active_modules ) ) {

            if ( self::is_available($id) ) {

                $active_modules[] = $id;

                //* Store altered modules
                update_option( 'ejo_base_active_modules', $active_modules );

                //* Allow code to be hooked here
                do_action( 'ejo_base_module_activation', $id );

                return true;
            }
        }

        return false;
    }

    public static function deactivate( $id, $active_modules = null )
    {
        if ( ! $active_modules )
            $active_modules = get_option( 'ejo_base_active_modules', array() );

        if ( self::is_activated( $id, $active_modules ) ) {

            //* Make sure all values occur once
            $active_modules = array_unique($active_modules);

            //* Remove module from modules
            $active_modules = array_unset_by_value( $id, $active_modules );

            //* Store altered modules
            update_option( 'ejo_base_active_modules', $active_modules );

            //* Allow code to be hooked here
            do_action( 'ejo_base_module_deactivation', $id );

            //* Check for dependancy
            self::check_activated_modules( $active_modules );

            return true;
        }

        return false;
    }

    /** 
     * Check currently activated modules 
     *  
     * Deactivate modules which shouldn't be activated (ie. missing dependancies)
     */
    public static function check_activated_modules( $active_modules = null )
    {
        if ( ! $active_modules )
            $active_modules = get_option( 'ejo_base_active_modules', array() );

        foreach ($active_modules as $id) {

            if ( ! self::is_available($id) ) {
                self::deactivate( $id, $active_modules );
            }
        }
    }
}

includes/options-page.php
<?php
if ( !current_user_can( 'manage_options' ) )  {
	wp_die( __( 'You do not have sufficient permissions to access this page.' ) );
}
?>

<style type="text/css">
	.bottom-buttons { clear: both; }
	.dependancy-good,
	.dependancies-hidden { color: #ddd; }
	.ejo-base-module.unavailable td{
		
	}
</style>

<div class="wrap">
	<h1><?php echo esc_html( get_admin_page_title() ); ?></h1> 

	<form action="<?php echo esc_attr( wp_unslash( $_SERVER['REQUEST_URI'] ) ); ?>" method="post">

		<table class="wp-list-table widefat plugins">
			<thead>
				<tr>
					<td id="cb" class="manage-column column-cb check-column"><?php /* <label class="screen-reader-text" for="cb-select-all-1">Alles selecteren</label><input id="cb-select-all-1" type="checkbox">*/ ?></td>
					<th scope="col" id="name" class="manage-column column-name column-primary">EJO Base Module</th>
					<th scope="col" id="description" class="manage-column column-description">Beschrijving</th>	
				</tr>
			</thead>

			<tbody id="the-list">

				<?php 

				$ejo_base_active_modules = get_option( 'ejo_base_active_modules', array() );
				foreach (EJO_Base_Module_Manager::$modules as $id => $module) {
					show_module_row( $module, $ejo_base_active_modules );
				}

				?>

			</tbody>

		</table>

	</form>

</div><!-- END .wrap -->
<?php

function show_module_row($module, $active_modules = null)
{
	if ( ! $active_modules )
            $active_modules = get_option( 'ejo_base_active_modules', array() );

	$menu_page = EJO_Base_Module_Manager::$slug;

	$is_available = EJO_Base_Module::is_available( $module['id'] );
	$has_theme_support = EJO_Base_Module::has_theme_support( $module['id'] );
	$is_active = EJO_Base_Module::is_active( $module['id'], $active_modules );

	$classes = ($has_theme_support) ? ' supported' : ' not-supported';
	$active = ($is_active) ? 'active' : 'inactive';
	$available = ($is_available) ? 'available' : 'unavailable';

	?>

	<tr class="<?php echo $active; ?> <?php echo $available; ?> ejo-base-module" data-slug="<?php echo $module['id']; ?>">

		<th scope="row" class="check-column">
			<label class="screen-reader-text" for="checkbox_<?php echo $module['id']; ?>"><?php echo $module['name']; ?> selecteren</label>
			<?php /* <input name="checked[]" value="<?php echo $module['id']; ?>" id="checkbox_<?php echo $module['id']; ?>" type="checkbox"> */ ?>
		</th>

		<td class="plugin-title column-primary">

			<strong><?php echo $module['name']; ?></strong>

			<div class="row-actions visible">

				<?php if ($is_available) : ?>

					<?php if ($is_active) : ?>
						<span class="deactivate">
							<a href="admin.php?page=<?php echo $menu_page; ?>&amp;action=deactivate&amp;module=<?php echo $module['id']; ?>" aria-label="<?php echo __('Deactivate') . ' ' . $module['name']; ?>"><?php _e('Deactivate'); ?></a>
						</span>
					<?php else : ?>
						<span class="activate">
							<a href="admin.php?page=<?php echo $menu_page; ?>&amp;action=activate&amp;module=<?php echo $module['id']; ?>" aria-label="<?php echo __('Activate') . ' ' . $module['name']; ?>"><?php _e('Activate'); ?></a>
						</span>
					<?php endif; // END active check ?>

				<?php else : ?>

					<span><?php EJO_Base_Module::print_why_not_available($module['id']); ?></span>

				<?php endif; // END availability check ?>

			</div>
			<button type="button" class="toggle-row">
				<span class="screen-reader-text"><?php _e('Show more details'); ?></span>
			</button>

		</td>
		<td class="column-description desc">
			<div class="plugin-description">
				<p><?php echo $module['description']; ?></p>
			</div>
			<div class="<?php echo $active; ?> second plugin-version-author-uri">

				<?php EJO_Base_Module::show_dependancies($module['id']); ?>

			</div>
		</td>
	</tr>
	<?php
}


function ejo_base_module_message( $message = 'Instellingen Opgeslagen' )
{
	echo '<div id="message" class="updated notice is-dismissible">';
	echo '<p>'.$message.'</p>';
	echo '<button type="button" class="notice-dismiss"><span class="screen-reader-text">Dit bericht verbergen.</span></button>';
	echo '</div>';
}



includes/modules/blog-comments.php
<?php

/**
 * Blog Comments
 */
if ( ! EJO_Base_Module::is_active('blog-comments') ) {

	/* Remove comments menu */
    add_action( 'admin_menu', function() {

        //* Manipulations do not count for admin users
        if (current_user_can('manage_options'))
            return;

        //* Loop through menu array
        foreach ($GLOBALS['menu'] as $index => $menu_item) {

	        //* edit-comments.php represents comments menu
	        if ($menu_item[2] == 'edit-comments.php' ) {

				// Unset top level menu
				unset( $GLOBALS['menu'][$index], $GLOBALS['submenu'][ 'edit-comments.php' ] );
				break;
			}
		}

    }, 1);

	//* Remove comments from admin-bar
	add_action( 'admin_bar_menu', function($wp_admin_bar) {

		//* Manipulations do not count for admin users
        if (current_user_can('manage_options'))
            return;

        $wp_admin_bar->remove_node('comments');      // Remove the comments link

    }, 99);

    //* Restrict access to comments screen
    add_action( 'current_screen', function($current_screen) {

        //* Manipulations do not count for admin users
        if (current_user_can('manage_options'))
            return;

        if ( 'edit-comments' == $current_screen->base ) {
            wp_die( __( 'You are not allowed to access the comments section. Contact your developer if this doesn\'t seem right.' ) );
        }
    });

    //* Remove comment-related dashboard widgets
    add_action('admin_init', function() {

        remove_meta_box( 'dashboard_recent_comments', 'dashboard', 'normal' );
    });

    //* Remove widget 
    add_action( 'widgets_init', function() {

    	//* Manipulations do not count for admin users
        if ( ! current_user_can('manage_options') )
            unregister_widget( 'WP_Widget_Recent_Comments' );
    }, 11);
}

ejo-base-module-manager.php
<?php
/**
 * Plugin Name:         EJO Base Module Manager
 * Plugin URI:          http://github.com/erikjoling/ejo-base-module-manager
 * Description:         Manages modules for my WordPress base websites
 * Version:             0.1
 * Author:              Erik Joling
 * Author URI:          http://www.ejoweb.nl/
 * Text Domain:         ejo-base-module-manager
 * Domain Path:         /languages
 *
 * GitHub Plugin URI:   https://github.com/erikjoling/ejo-base-module-manager
 * GitHub Branch:       master
 *
 * Minimum PHP version: 5.3.0
 *
 * @package   EJO Base Module Manager
 * @version   0.1.0
 * @since     0.1.0
 * @author    Erik Joling <erik@ejoweb.nl>
 * @copyright Copyright (c) 2015, Erik Joling
 * @link      http://github.com/erikjoling
 */

/**c
 *
 */
final class EJO_Base_Module_Manager 
{
    /* Holds the instance of this class. */
    private static $_instance = null;

    /* Version number of this plugin */
    public static $version = '0.1';

    /* Store the slug of this plugin */
    public static $slug = 'ejo-base-module-manager';

    /* Stores the directory path for this plugin. */
    public static $dir;

    /* Stores the directory URI for this plugin. */
    public static $uri;

    /* Stores the includes directory path for this plugin. */
    public static $inc_dir;

    /* Stores activated modules */
    public static $modules = array();

    /* Only instantiate once */
    public static function init() 
    {
        if ( !self::$_instance )
            self::$_instance = new self;
        return self::$_instance;
    }

    //* No cloning
    private function __clone() {}

    /* Plugin setup. */
    private function __construct() 
    {
        //* Setup common plugin stuff
        self::plugin_setup();

        //* Set modules
        self::set_modules();

        //* Immediatly include helpers
        add_action( 'plugins_loaded', array( 'EJO_Base_Module_Manager', 'helpers' ), 9 );

        //* Immediatly include helpers
        add_action( 'plugins_loaded', array( 'EJO_Base_Module_Manager', 'module_manager' ) );
    }

    
    /* Defines the directory path and URI for the plugin. */
    public static function plugin_setup() 
    {
        self::$dir = plugin_dir_path( __FILE__ );
        self::$inc_dir = self::$dir . 'includes/';
        self::$uri = plugin_dir_url( __FILE__ );

        /* Load the translation for the plugin */
        load_plugin_textdomain( 'ejo-base-module-manager', false, 'ejo-base-module-manager/languages' );
    }

    /* Add helper functions */
    public static function helpers() 
    {
        /* Include helpers */
        require_once( self::$inc_dir . 'helpers.php' );

        /* Include class module */
        require_once( self::$inc_dir . 'class-module.php' );
    }

    /* Add helper functions */
    public static function module_manager() 
    {
        /* Allow array-arguments to be passed for theme-support:ejo-base-modules */
        add_filter( 'current_theme_supports-ejo-base-modules', 'ejo_add_extended_theme_support', 10, 3 );
        
        /* Add EJObase Option page to Wordpress Option menu */
        add_action( 'admin_menu', array( 'EJO_Base_Module_Manager', 'register_menu' ), 1 );

        //* Save Activate/Deactivate actions of options page
        add_action( 'after_setup_theme', array( 'EJO_Base_Module_Manager', 'save_module_activations' ), 98 );

        //* Hook to module activation and deactivation
        add_action( 'ejo_base_module_activation', array( 'EJO_Base_Module_Manager', 'reset_caps_on_module_activation' ) ); 
        add_action( 'ejo_base_module_deactivation', array( 'EJO_Base_Module_Manager', 'reset_caps_on_module_activation' ) );

        //* Check modules after plugin (de)activations
        add_action( 'after_setup_theme', array( 'EJO_Base_Module_Manager', 'check_modules_on_every_plugin_activation' ), 98 );

        //* Code to activate when some modules are (in)active (hook after `save_module_activations`)
        add_action( 'after_setup_theme', array( 'EJO_Base_Module_Manager', 'module_manipulations' ), 99 );
    }

    /**
     * Process the actions of the EJO Base Modules Options Page
     */
    public static function save_module_activations()
    {
        //* Perform action if set
        if ( isset($_GET['action']) && isset($_GET['module']) ) {

            $module_id = esc_attr($_GET['module']);

            if ($_GET['action'] == 'activate') {

                if ( EJO_Base_Module::activate( $module_id ) ) {
                    add_action( 'admin_notices', array( 'EJO_Base_Module_Manager', 'show_activation_message' ) );
                }
            } 

            elseif ($_GET['action'] == 'deactivate') {

                if ( EJO_Base_Module::deactivate( $module_id ) ) {
                    add_action( 'admin_notices', array( 'EJO_Base_Module_Manager', 'show_deactivation_message' ) );
                }
            }
        }
    }

    private static function set_modules()
    {
        EJO_Base_Module_Manager::$modules = array(
            'blog' => array(
                'id'           => 'blog',
                'name'         => __( 'Blog', EJO_Base_Module_Manager::$slug ),
                'description'  => __( 'Overzichtspagina, artikelpagina en widget met laatste artikelen. Voor bloggen of nieuws.', EJO_Base_Module_Manager::$slug ),
                'dependancies' => array(),
            ),
            'blog-comments' => array(
                'id'           => 'blog-comments',
                'name'         => __( 'Blog Comments', EJO_Base_Module_Manager::$slug ),
                'description'  => __( 'Mogelijkheid voor bezoekers om reacties achter te laten onder uw blogartikelen', EJO_Base_Module_Manager::$slug ),
                'dependancies' => array(
                    array(
                        'type' => 'ejo-base-module',
                        'id'   => 'blog',
                    ),
                ),
            ),
            'testimonials' => array(
                'id'           => 'testimonials',
                'name'         => __( 'Testimonials', EJO_Base_Module_Manager::$slug ),
                'description'  => __( 'Overzichtspagina met referenties en widget met laatste referenties', EJO_Base_Module_Manager::$slug ),
                'dependancies' => array(
                    array(
                        'type'  => 'plugin',
                        'name'  => 'EJO Simple Testimonials',
                        'class' => 'EJO_Simple_Testimonials',
                    ),
                ),
            ),
            'portfolio' => array(
                'id'           => 'portfolio',
                'name'         => __( 'Portfolio', EJO_Base_Module_Manager::$slug ),
                'description'  => __( 'Overzichtspagina met projecten en widget met laatste projecten', EJO_Base_Module_Manager::$slug ),
                'dependancies' => array(
                    array(
                        'type'  => 'plugin',
                        'name'  => 'EJO Portfolio',
                        'class' => 'EJO_Portfolio',
                    ),
                ),
            ),
        );
    }

    public static function show_activation_message()
    {
        $class = 'notice updated is-dismissible';
        $message = 'Module <strong>' . __('activated') . '</strong>';

        printf( '<div class="%1$s"><p>%2$s</p></div>', $class, $message ); 
    }

    public static function show_deactivation_message()
    {
        $class = 'notice updated is-dismissible';
        $message = 'Module <strong>' . __('deactivated') . '</strong>';

        printf( '<div class="%1$s"><p>%2$s</p></div>', $class, $message ); 
    }

    /* Register EJObase Options Menu Page */
    public static function register_menu()
    {
        add_submenu_page( 'options-general.php', __('EJO Base Modules'), __('EJO Base Modules'), 'manage_options', self::$slug, array( 'EJO_Base_Module_Manager', 'add_menu_page' ) );
    }

    /* Add EJObase Options Menu Page */
    public static function add_menu_page()
    {
        /* Include theme options page */
        require_once( self::$inc_dir . 'options-page.php' );
    }

    /** 
     * Module Manipulations 
     *
     * Must be hooked after `check_modules_on_every_plugin_activation`
     * Because it needs to check if modules are active
     * And `check_modules_on_every_plugin_activation` impacts that
     */
    public static function module_manipulations()
    {
        //* Modules
        require_once( self::$inc_dir . 'modules/blog.php' ); // Blog
        require_once( self::$inc_dir . 'modules/blog-comments.php' ); // Blog Comments
        require_once( self::$inc_dir . 'modules/testimonials.php' ); // EJO Simple testimonials
        // require_once( self::$inc_dir . 'modules/portfolio.php' ); // EJO Portfolio
        // require_once( self::$inc_dir . 'modules/popup-box.php' ); // EJO Popup-box
        // require_once( self::$inc_dir . 'modules/photo-gallery.php' ); // EJO Photo Gallery
        // require_once( self::$inc_dir . 'modules/team.php' ); // EJO Team
        // require_once( self::$inc_dir . 'modules/social-media-extra.php' ); // EJO Social Media Pack
        // require_once( self::$inc_dir . 'modules/faq.php' ); // FAQ
    }

    /* On module activation */
    public static function reset_caps_on_module_activation( $module_id )
    {
        //* Hook to end of admin init to ensure all module manipulations and checks are done
        add_action( 'admin_init', array( 'EJO_Base_Module_Manager', 'reset_client_caps') );
    }

    /* On module activation */
    public static function check_modules_on_every_plugin_activation()
    {
        global $pagenow;

        if ($pagenow == 'plugins.php') {

            if ( isset($_GET['activate']) || isset($_GET['deactivate']) || isset($_GET['activate-multi']) || isset($_GET['deactivate-multi']) ) {
                EJO_Base_Module::check_activated_modules();
            }
        }
    }

    /* Reset the caps of the client-role */
    public static function reset_client_caps()
    {
        if ( class_exists('EJO_Client') ) {

            EJO_Client::reset_client_caps();

            /**
             * Remove double client-cap-reset
             *
             * Situation: 
             * - EJO_Client reset caps on every plugin (de)activation
             * - When a plugin is deactivated which causes a module to deactivate this class will run a EJO_Client reset cap
             * - No need to run two client cap resets
             */
            remove_action( 'admin_init', array( 'EJO_Client', 'reset_on_every_plugin_activation'), 99);
        }
    }
}

/* Call EJO Base Module Manager */
EJO_Base_Module_Manager::init();
